src/core/reflection/sobject/reflect-custom-field-source.ts

```diff
@@ -58,12 +59,34 @@ function validate(parsedResult: unknown): E.Either<Error, { CustomField: unknown
 }
 
 function toCustomFieldMetadata(parserResult: { CustomField: unknown }): CustomFieldMetadata {
-  const customField = typeof parserResult.CustomField === 'object' ? parserResult.CustomField : {};
+  const customField =
+    parserResult?.CustomField != null && typeof parserResult.CustomField === 'object' ? parserResult.CustomField : {};
   const defaultValues = {
     description: null,
   };
 
-  return { ...defaultValues, ...customField, type_name: 'customfield' } as CustomFieldMetadata;
+  const pickListValues =
+    isCustomField(customField) && customField.type === 'Picklist' ? toPickListValues(customField) : [];
+  return { ...defaultValues, ...customField, type_name: 'customfield', pickListValues } as CustomFieldMetadata;
+}
+
+function toPickListValues(customField: object): string[] {
+  if ('valueSet' in customField) {
+    const valueSet = customField.valueSet as object;
+    if ('valueSetDefinition' in valueSet) {
+      const valueSetDefinition = valueSet.valueSetDefinition as object;
+      if ('value' in valueSetDefinition) {
+        const pickListValues = valueSetDefinition.value as object[];
+        return pickListValues.filter((each) => 'fullName' in each).map((each) => each.fullName as string);
+      }
+    }
+  }
+
+  return [];
+}
+
+function isCustomField(customField: object): customField is CustomFieldMetadata {
```

Comment:
I'm wondering if this should be renamed to something like `hasType`, since at this stage we already know we are dealing with custom field metadata because of the previous check (`validate`)